src/components/FooterFiltering/FooterFiltering.tsx

```diff
@@ -0,0 +1,73 @@
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  filter: string;
+  setFilter: (a: string) => void;
+  todos: Todo[];
+};
+
+export const FooterFiltering: React.FC<Props> = ({
+  filter,
+  setFilter,
+  todos,
+}) => {
+  function handleFilter(event: React.MouseEvent<HTMLElement>) {
+    setFilter(event.currentTarget.textContent || '');
```

Comment:
in case you use the union type for `filter` state you should use typecasting here